src/actix.rs

```diff
@@ -0,0 +1,118 @@
+//! Actix response types for [`ProblemDetails`].
+//!
+//! Requires feature `actix`.
+//!
+//! With the `actix` feature enabled, [`ProblemDetails`] implements [`ResponseError`] using
+//! [`JsonProblemDetails`]. You can also return [`JsonProblemDetails`] to be specific.
+//! If you want to return XML, you can use [`XmlProblemDetails`].
+//!
+//! # Example
+//!
+//! ```rust
+//! use actix_web::{App, web, HttpServer};
+//! use http::StatusCode;
+//! use problem_details::ProblemDetails;
+//!
+//! async fn handler() -> Result<&'static str, ProblemDetails> {
+//!     // always return a problem description
+//!     Err(ProblemDetails::from_status_code(StatusCode::IM_A_TEAPOT)
+//!         .with_detail("short and stout"))
+//! }
+//!
+//! fn main() {
+//!     HttpServer::new(|| {
+//!         App::new()
+//!             .route("/", web::get().to(handler))
+//!     // build and run server...
+//! }
+//! ```
+use actix_web::{
+    web::Json,
+    {HttpResponse, ResponseError}
+};
+use std::fmt::Debug;
+use http::StatusCode;
+
+use crate::ProblemDetails;
+
+#[cfg(feature = "json")]
+use crate::JsonProblemDetails;
+
+#[cfg(feature = "xml")]
+use crate::XmlProblemDetails;
+
+#[cfg(feature = "json")]
+impl<Ext> ResponseError for ProblemDetails<Ext>
+where
+    Ext: serde::Serialize + Clone + Send + Debug,
+{
+    fn error_response(&self) -> HttpResponse {
+        self.clone().into()
```

Comment:
We can work without the clone here, since `&ProblemDetails` implements serde::Serialize as well.

```suggestion
        let status_code = self.status_code();
        let content = self;

        HttpResponse::build(status_code)
            .content_type(JsonProblemDetails::<Ext>::CONTENT_TYPE)
            .json(Json(content))
```

Reply:
Do you mean we don't need From trait here? 
Please look at my last commit when you can.